pages/_app.js

```diff
@@ -88,6 +89,7 @@ export default function App({ Component, pageProps }) {
         <Component
           onAddEmotionDetails={handleAddEmotionDetails}
           emotionEntries={emotionEntries}
+          setEmotionEntries={setEmotionEntries}
```

Comment:
**Try to avoid passing react setter functions directly as props:** 

This can lead to potentially error-prone code, especially as applications grow in complexity.  A more robust way is to wrapping setter functions in handler functions before passing them to child components. **Just as you have done so far.** 🙂

Consider to encapsulating the state update logic for the restoring from backup in a function inside the `_app.js `. Then, in my opinion, you don't need to drill the setter function down. 

Example function to handle setting entries from backup

```js
function  restoreFromBackup (){
        setEmotionEntries([...backupEntries]);
    };
```

> ⚠️ Assuming `shownEntries` in `DataGenerator.js` is intended to display what's currently active, this can be the same.